packages/roosterjs-editor-dom/lib/utils/tableFormatInfo.ts

```diff
@@ -0,0 +1,111 @@
+import { TableBorderFormat, TableFormat } from 'roosterjs-editor-types';
+
+const TABLE_STYLE_INFO = 'roosterTableInfo';
+const DEFAULT_FORMAT: TableFormat = {
+    topBorderColor: '#ABABAB',
+    bottomBorderColor: '#ABABAB',
+    verticalBorderColor: '#ABABAB',
+    hasHeaderRow: false,
+    hasFirstColumn: false,
+    hasBandedRows: false,
+    hasBandedColumns: false,
+    bgColorEven: null,
+    bgColorOdd: '#ABABAB20',
+    headerRowColor: '#ABABAB',
+    tableBorderFormat: TableBorderFormat.DEFAULT,
+};
+
+/**
+ * Get the format info of a table
+ * If the table does not have a info saved, it will be retrieved from the css styles
+ * @param table The table that has the info
+ */
+export function getTableFormatInfo(table: HTMLTableElement) {
+    const obj = safeParseJSON(table?.dataset[TABLE_STYLE_INFO]) as TableFormat;
+    return checkIfTableFormatIsValid(obj) ? obj : DEFAULT_FORMAT;
+}
+
+function checkIfTableFormatIsValid(format: TableFormat) {
+    if (!format) {
+        return false;
+    }
+    const {
+        topBorderColor,
+        verticalBorderColor,
+        bottomBorderColor,
+        bgColorOdd,
+        bgColorEven,
+        hasBandedColumns,
+        hasBandedRows,
+        hasFirstColumn,
+        hasHeaderRow,
+        tableBorderFormat,
+    } = format;
+    const colorsValues = [
+        topBorderColor,
+        verticalBorderColor,
+        bottomBorderColor,
+        bgColorOdd,
+        bgColorEven,
+    ];
+    const stateValues = [hasBandedColumns, hasBandedRows, hasFirstColumn, hasHeaderRow];
+
+    if (
+        colorsValues.some(key => !isAValidType(key)) ||
+        stateValues.some(key => !isBoolean(key)) ||
+        !isAValidTableBorderType(tableBorderFormat)
+    ) {
+        return false;
+    }
+
+    return true;
+}
+
+function isAValidType(a: any) {
+    if (a === null || a === undefined || typeof a === 'string') {
+        return true;
+    }
+    return false;
+}
```

Comment:
If possible add a name with more defintion other than `a`, maybe something like colorInput.
Name of the function could be isAValidColor.

colorValues is a string[], so I think we could change the parameter type to string or is there a specific scenario where is going to be a different type?